service/src/main/java/bio/terra/cbas/dependencies/bard/BardService.java

```diff
@@ -0,0 +1,82 @@
+package bio.terra.cbas.dependencies.bard;
+
+import bio.terra.bard.api.DefaultApi;
+import bio.terra.bard.client.ApiClient;
+import bio.terra.bard.model.EventsEvent200Response;
+import bio.terra.bard.model.EventsEventLogRequest;
+import bio.terra.cbas.dependencies.common.HealthCheck;
+import bio.terra.cbas.model.RunSetRequest;
+import bio.terra.cbas.models.GithubMethodDetails;
+import bio.terra.cbas.models.MethodVersion;
+import bio.terra.common.iam.BearerToken;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class BardService implements HealthCheck {
+
+  Logger log = LoggerFactory.getLogger(BardService.class);
+
+  private final BardClient bardClient;
+  private final String appId = "cbas";
+
+  public BardService(BardClient bardClient) {
+    this.bardClient = bardClient;
+  }
+
+  private DefaultApi getDefaultApi(BearerToken userToken) {
+    ApiClient client = bardClient.bardAuthClient(userToken);
+    return bardClient.defaultApi(client);
+  }
+
+  public void logRunSetEvent(
+      RunSetRequest request,
+      MethodVersion methodVersion,
+      GithubMethodDetails githubMethodDetails,
+      List<String> workflowIds,
+      BearerToken userToken) {
+    String eventName = "workflow-submission";
+    HashMap<String, String> properties = new HashMap<>();
+    properties.put("runSetName", request.getRunSetName());
+    properties.put("methodName", methodVersion.method().name());
+    properties.put("methodSource", methodVersion.method().methodSource());
+    properties.put("methodVersionName", methodVersion.name());
+    properties.put("methodVersionUrl", methodVersion.url());
+    properties.put("recordCount", String.valueOf(request.getWdsRecords().getRecordIds().size()));
```

Comment:
I'm inclined to remove this now that we have the Cromwell workflow ids. Is there a reason to log how many records a user wanted to analyze?

Reply:
couldnt hurt to leave there imo

Reply:
In the future we might no longer have a 1:1 mapping between WDS rows and Cromwell workflows, eg:

- Zip together entries from 2 different 10-column tables to produce 10 workflows from 20 WDS records
- Cross product entries from 2 different 10-column tables to produce 100 workflows from 20 WDS records
- Allow non-WDS backed workflows to run, producing 1 (or many) workflows from 0 records

... so I think the more we record, and the more generic the naming can be, the better